test.js

```diff
@@ -0,0 +1,597 @@
+    })
+
+    req.on('close', () => t.pass('server request closed'))
+    res.on('close', () => t.pass('server response closed'))
+  })
+
+  server.listen(0)
+  await waitForServer(server)
+
+  const req = http.request({
+    method: 'GET',
+    host: server.address().address,
+    port: server.address().port,
+    path: '/'
+  }, function (res) {
+    t.is(res.statusCode, 200)
+
+    const chunks = []
+    res.on('data', (chunk) => chunks.push(chunk))
+    res.on('end', () => {
+      const body = b4a.concat(chunks)
+      t.alike(body, b4a.from('part 1 + part 2'), 'client response ended')
+    })
+  })
+
+  req.on('close', () => {
+    t.pass('client request closed')
+    server.close()
+  })
+
+  req.on('error', (err) => t.fail('client req error: ' + err.message + ' (' + err.code + ')'))
+
+  req.end()
+})
+
+// + solo this test and it would randomly fail
+test('server does a big write', async function (t) {
+  t.plan(7)
+
+  const server = createServer()
+  server.on('close', () => t.pass('server closed'))
+
+  server.on('connection', function (socket) {
+    socket.on('close', () => t.pass('server socket closed'))
+    socket.on('error', (err) => {
+      console.error(err)
+      t.fail('server socket error: ' + err.message + ' (' + err.code + ')')
+    })
+  })
+
+  server.on('request', function (req, res) {
+    res.write(b4a.alloc(2 * 1024 * 1024, 'abcd'))
+    setImmediate(() => {
+      res.end(b4a.alloc(2 * 1024 * 1024, 'efgh'))
+    })
+
+    req.on('close', () => t.pass('server request closed'))
+    res.on('close', () => t.pass('server response closed'))
+  })
+
+  server.listen(0)
+  await waitForServer(server)
+
+  const req = http.request({
+    method: 'GET',
+    host: server.address().address,
+    port: server.address().port,
+    path: '/'
+  }, function (res) {
+    t.is(res.statusCode, 200)
+
+    const chunks = []
+    res.on('data', (chunk) => chunks.push(chunk))
+    res.on('end', () => {
+      const body = b4a.concat(chunks)
+      const expected = b4a.concat([b4a.alloc(2 * 1024 * 1024, 'abcd'), b4a.alloc(2 * 1024 * 1024, 'efgh')])
+      t.alike(body, expected, 'client response ended')
+    })
+  })
+
+  req.on('close', () => {
+    t.pass('client request closed')
+    server.close()
+  })
+
+  req.on('error', (err) => t.fail('client req error: ' + err.message + ' (' + err.code + ')'))
+
+  req.end()
+})
+
+/* test.solo('client does a small write', async function (t) {
+  t.plan(8)
+
+  const server = createServer()
+  server.on('close', () => t.pass('server closed'))
+
+  server.on('connection', function (socket) {
+    socket.on('close', () => t.pass('server socket closed'))
+    socket.on('error', (err) => {
+      console.error(err)
+      t.fail('server socket error: ' + err.message + ' (' + err.code + ')')
+    })
+  })
+
+  server.on('request', function (req, res) {
+    const chunks = []
+    res.on('data', (chunk) => {
+      t.comment('res data received: ' + chunk.byteLength)
```

Comment:
How do I consume i.e. post data from server side?

Reply:
That `res.on('data', cb)` is not being called, or maybe I did wrong the request?

Reply:
streaming client bodies not supported, so just kill this test